cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/repo.go

```diff
@@ -48,21 +48,40 @@ var (
 )
 
 // namespace maybe apiv1.NamespaceAll, in which case repositories from all namespaces are returned
+// the repositories to which the caller context has no access are filtered out
 func (s *Server) listReposInNamespace(ctx context.Context, namespace string) ([]sourcev1.HelmRepository, error) {
-	client, err := s.getClient(ctx, namespace)
+	// the actual List(...) call will be executed in the context of
+	// kubeapps-internal-kubeappsapis service account
+	// ref https://github.com/kubeapps/kubeapps/issues/4390 for explanation
```

Comment:
I would expect this to only be if the request is across all namespaces, otherwise it's going to be simpler/safer to request the repos in a single namespace with the user context?

Reply:
np. I will make that change

Reply:
Not sure if you misread my comment, but I wasn't meaning that this function should change to list all repos, but rather that we'd only be listing all repos if the request was for `apiv1.NamespaceAll`, otherwise if the function was meant to list repos in a specific namespace, we should use the user client (not service account). But that said, maybe you have a reason for changing it like this (maybe there's no need to list repos just in a specific namespace, not sure).

Reply:
Eeeh, on the 2nd look, that's not right. Even when the caller asks `GetAvailablePackagesSummaries` for namespace-A, flux plugin should return packages from repos regardless which namespace they are in (of course minus repos in namespaces caller has no access to). These are all packages available for install as far as flux is concerned. So I think the `List() `request should always be cluster-wide. There is no use case I can see where the `List()` call is done w.r.t. to a single namespace.